src/sas/qtgui/Plotting/Plotter2D.py

```diff
@@ -309,6 +311,46 @@ def getActivePlots(self):
         ''' utility method for manager query of active plots '''
         return self.manager.active_plots
 
+    def getActiveSlicerPlots(self):
+        """
+        Utility method for manager query of active slicer plots.
+        Returns a dictionary of plot_id: plot_window for all 1D plots
+        that were created by slicers from this 2D plot.
+        """
+        slicer_plots = {}
+
+        if not hasattr(self, 'manager') or not hasattr(self.manager, 'active_plots'):
+            return slicer_plots
+
+        # Search through all active plots
+        for plot_id, plot_window in self.manager.active_plots.items():
+            if hasattr(plot_window, 'data') and plot_window.data is not None:
```

Comment:
To save some indentation later, a suggestion (though what plot window doesn't have the data attr?):

```
if not hasattr(plot_window, 'data') or not plot_window.data:
    # No data loaded (catches the cases where data == None and data == [])
    continue

data_list = ___
...
```